File: algorithms/geometry/point-in-polygon/go/point_in_polygon.go
package pointinpolygon

func PointInPolygon(arr []int) int {
	px, py := arr[0], arr[1]
	n := arr[2]

	inside := false
	j := n - 1
	for i := 0; i < n; i++ {
		xi, yi := arr[3+2*i], arr[3+2*i+1]
		xj, yj := arr[3+2*j], arr[3+2*j+1]

		if (yi > py) != (yj > py) &&
			float64(px) < float64(xj-xi)*float64(py-yi)/float64(yj-yi)+float64(xi) {
			inside = !inside
		}
		j = i
	}

	if inside {
		return 1
	}
	return 0
}

File: algorithms/geometry/point-in-polygon/csharp/PointInPolygon.cs
using System;

public class PointInPolygon
{
    public static int CheckPointInPolygon(int[] arr)
    {
        double px = arr[0], py = arr[1];
        int n = arr[2];

        bool inside = false;
        int j = n - 1;
        for (int i = 0; i < n; i++)
        {
            double xi = arr[3 + 2 * i], yi = arr[3 + 2 * i + 1];
            double xj = arr[3 + 2 * j], yj = arr[3 + 2 * j + 1];

            if ((yi > py) != (yj > py) &&
                px < (xj - xi) * (py - yi) / (yj - yi) + xi)
            {
                inside = !inside;
            }
            j = i;
        }

        return inside ? 1 : 0;
    }
}
